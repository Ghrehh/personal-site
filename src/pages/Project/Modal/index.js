// @flow
import * as React from 'react';

import * as pathConstants from 'constants/paths';
import Page from 'components/Page';
import PageHeading from 'components/PageHeading';
import TransparentButton from 'components/TransparentButton';
import UnstyledLink from 'components/UnstyledLink';
import SlideUp from 'components/animation/SlideUp';

import styles from './styles.module.css';

type Props = {
  backgroundColor: string,
  heading: string,
  children: React.Node
};

const Modal = (props: Props) => (
  <SlideUp>
    <Page
      className={styles.page}
      style={{ backgroundColor: props.backgroundColor }}
      displayFooter={false}
    >
      <div className={styles.closeButtonContainer}>
        <TransparentButton>
          <UnstyledLink link={pathConstants.PROJECTS}>
            <span className={styles.closeButton}>Ã—</span>
          </UnstyledLink>
        </TransparentButton>
      </div>

      <PageHeading className={styles.title}>{props.heading}</PageHeading>

      {props.children}
    </Page>
  </SlideUp>
);

export default Modal;
