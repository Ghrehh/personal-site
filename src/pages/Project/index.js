// @flow
import React from 'react';

import * as projectsModel from 'models/data/projects';
import * as projectRouterParams from 'models/project-router-params';
import * as pathConstants from 'constants/paths';
import Page from 'components/Page';
import PageHeading from 'components/PageHeading';
import SectionHeading from 'components/SectionHeading';
import LinkButton from 'components/LinkButton';
import TransparentButton from 'components/TransparentButton';
import UnstyledLink from 'components/UnstyledLink';
import ButtonContainer from 'components/ButtonContainer';

import styles from './styles.module.css';

type Props = {
  projects: projectsModel.Type,
  match: projectRouterParams.Type
};

class Project extends React.Component<Props> {
  project() {
    return this.props.projects[this.props.match.params.project];
  }

  renderWebsiteButton() {
    // assignment necessary as flow doesn't understand the function will always return the
    // same value.
    const link = this.project().websiteLink;
    if (link === undefined) return null;

    return (
      <LinkButton external={true} link={link}>
        Website
      </LinkButton>
    );
  }

  render() {
    return (
      <Page className={styles.page} style={{ backgroundColor: this.project().backgroundColor }}>
        <div className={styles.closeButtonContainer}>
          <TransparentButton>
            <UnstyledLink link={pathConstants.PROJECTS}>
              <span className={styles.closeButton}>
                Ã—
              </span>
            </UnstyledLink>
          </TransparentButton>
        </div>
        <PageHeading className={styles.title}>
          {this.project().name}
        </PageHeading>

        <section className={styles.description}>
          <SectionHeading>Description</SectionHeading>
          <p className={styles.body}>{this.project().description}</p>
        </section>

        <section className={styles.links}>
          <SectionHeading>Links</SectionHeading>

          <ButtonContainer className={styles.linkButtonContainer}>
            <LinkButton external={true} link={this.project().gitHubLink}>
              GitHub
            </LinkButton>
            {this.renderWebsiteButton()}
          </ButtonContainer>
        </section>
      </Page>
    );
  }
}

export default Project;
