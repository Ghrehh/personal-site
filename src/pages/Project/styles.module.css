.page {
  padding-top: 0;
}

.closeButtonContainer {
  text-align: right;
  margin-bottom: 10px;
}

.closeButton {
  color: var(--charcoal);
  font-size: 50px;
  padding-right: 13px;
  transition: filter 0.5s;
}

.closeButton:hover {
  filter: brightness(1.5);
}

.body {
  text-align: center;
}

.description {
  margin-bottom: 40px;
  max-width: 500px;
  margin: auto;
  padding: 0 20px;
}

.links {
  margin-bottom: 40px;
}

.linkButtonContainer {
  text-align: center;
}
