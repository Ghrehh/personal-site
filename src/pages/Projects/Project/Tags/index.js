// @flow
import React from 'react';

import styles from './styles.module.css';

type Props = {
  tags: string[]
};

const Tags = (props: Props) => (
  <ul className={styles.tags}>
    {props.tags.map(tag => <li className={styles.tag}>{tag}</li>)}
  </ul>
);

export default Tags;
