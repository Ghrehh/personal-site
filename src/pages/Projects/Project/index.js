// @flow
import React from 'react';

import * as projectModel from 'models/data/project';
import * as pathConstants from 'constants/paths';
import SectionHeading from 'components/text/SectionHeading';
import UnstyledLink from 'components/UnstyledLink';

import styles from './styles.module.css';

type Props = {
  project: projectModel.Type,
  path: string
};

const Project = (props: Props) => (
  <div className={styles.projectOuter}>
    <UnstyledLink link={`${pathConstants.PROJECTS}/${props.path}`}>
      <div
        style={{ backgroundColor: props.project.backgroundColor }}
        className={styles.project}
      >
        <div className={styles.projectInner}>
          <SectionHeading className={styles.projectName}>
            {props.project.name}
          </SectionHeading>

          <p className={styles.projectBody}>{props.project.description}</p>
        </div>
      </div>
    </UnstyledLink>
  </div>
);

export default Project;
