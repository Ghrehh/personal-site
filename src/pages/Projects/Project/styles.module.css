.project {
  margin: 40px 20px 20px;
  transition: transform 0.5s, filter 0.5s;
}

.project:hover {
  cursor: pointer;
  transform: translate(0, -4px);
  filter: brightness(1.1);
}

.link {
  text-decoration: none;
}

.projectInner {
  max-width: 500px;
  margin: auto;
  padding: 40px;
  border-radius: 3px;
  background-color: white;
  text-align: center;
  box-shadow: var(--box-shadow);
}

.projectName {
  color: white;
}

.projectBody {
  margin: 0;
  color: white;
}

@media (max-width: 530px) {
  .project {
    margin: 0;
  }

  .project:hover {
    transform: initial;
  }

  .projectInner {
    box-shadow: none;
    border-radius: 0;
  }

}

