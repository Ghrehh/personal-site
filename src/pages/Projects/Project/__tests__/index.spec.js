import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Project from '../';

describe('Project', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};

    const project = { name: 'foo', description: 'bar', backgroundColor: 'red' };

    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(<Project project={project} />);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
