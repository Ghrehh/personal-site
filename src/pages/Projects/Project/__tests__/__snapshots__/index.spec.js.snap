// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project renders the component correctly 1`] = `
<div
  className="project"
>
  <LinkButton
    external={false}
    link="/projects/undefined"
  >
    <div
      className="projectInner"
      style={
        Object {
          "backgroundColor": "red",
        }
      }
    >
      <SectionHeading
        className="projectName"
      >
        foo
      </SectionHeading>
      <p
        className="projectBody"
      >
        bar
      </p>
    </div>
  </LinkButton>
</div>
`;
